core/controllers/base_test.py

```diff
@@ -303,38 +307,51 @@ def test_valid_pillow_path(self):
             Exception, 'Invalid path for oppia_tools library: invalid_path')
 
         def mock_os_path_join_for_pillow(*args):
+            """Mocks path for Pillow to test against an invalid path."""
             path = ''
             if args[1] == 'Pillow-6.0.0':
                 return 'invalid_path'
             else:
                 for arg in args:
-                    path += arg + '/'
+                    path = '/'.join(args)
                 return path
 
         pil_path_swap = self.swap(os.path, 'join', mock_os_path_join_for_pillow)
+        # We need to delete the existing module else the re-importing
+        # would just call the existing module.
         del sys.modules['appengine_config']
 
         with assert_raises_regexp_context_manager, pil_path_swap:
+            # This pragma is needed since we are re-importing under
+            # invalid conditions.
             import appengine_config  # pylint: disable-all
 
     def test_valid_third_party_library_path(self):
         import appengine_config
         assert_raises_regexp_context_manager = self.assertRaisesRegexp(
             Exception, 'Invalid path for third_party library: invalid_path')
 
-        def mock_os_path_join_for_pillow(*args):
+        def mock_os_path_join_for_third_party_lib(*args):
+            """Mocks path for third_party libs to test against an invalid
+            path.
+            """
             path = ''
             if args[1] == 'third_party':
                 return 'invalid_path'
             else:
                 for arg in args:
-                    path += arg + '/'
+                    path = '/'.join(args)
                 return path
 
-        pil_path_swap = self.swap(os.path, 'join', mock_os_path_join_for_pillow)
+        pil_path_swap = self.swap(
```

Comment:
Why is pillow invoked here (in the variable name)?

Please proofread your PR more carefully and make sure everything is correct and fully explained. Otherwise, this is leading to a lot of back-and-forth.